pkg/autodiscovery/common/utils/kubelet.go

```diff
@@ -16,3 +20,35 @@ func GetCustomCheckID(annotations map[string]string, containerName string) (stri
 	id, found := annotations[fmt.Sprintf(checkIDAnnotationFormat, containerName)]
 	return id, found
 }
+
+// ValidateAnnotationsMatching detects if AD annotations don't match a valid container identifier
+func ValidateAnnotationsMatching(annotations map[string]string, containerIdentifiers map[string]bool, containerNames map[string]bool, adPrefix string) []error {
+	var errors []error
+	adAnnotation := fmt.Sprintf(`%s.+\..+`, adPrefix)
```

Comment:
Is `adPrefix` supposed to be interpreted as a regular expression here?

Reply:
Yes, it's being used to build a regular expression (`ad.datadoghq.com/.+\..+`). I'll work on making that clearer